core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -298,61 +277,83 @@ public ProtoStreamWriterImpl getProtoStreamWriter() {
+      protected void writeUInt32Field(int fieldNumber, int value) throws IOException {
          writeVarint32(WireType.makeTag(fieldNumber, WireType.WIRETYPE_VARINT));
          writeVarint32(value);
       }
 
-      void writeUInt64Field(int fieldNumber, long value) throws IOException {
+      protected void writeUInt64Field(int fieldNumber, long value) throws IOException {
          writeVarint32(WireType.makeTag(fieldNumber, WireType.WIRETYPE_VARINT));
          writeVarint64(value);
       }
 
-      void writeFixed32Field(int fieldNumber, int value) throws IOException {
+      protected void writeFixed32Field(int fieldNumber, int value) throws IOException {
          writeVarint32(WireType.makeTag(fieldNumber, WireType.WIRETYPE_FIXED32));
          writeFixed32(value);
       }
 
-      void writeFixed64Field(int fieldNumber, long value) throws IOException {
+      protected void writeFixed64Field(int fieldNumber, long value) throws IOException {
          writeVarint32(WireType.makeTag(fieldNumber, WireType.WIRETYPE_FIXED64));
          writeFixed64(value);
       }
 
-      void writeBoolField(int fieldNumber, boolean value) throws IOException {
+      protected void writeBoolField(int fieldNumber, boolean value) throws IOException {
          writeVarint32(WireType.makeTag(fieldNumber, WireType.WIRETYPE_VARINT));
          writeByte((byte) (value ? 1 : 0));
       }
 
-      void writeLengthDelimitedField(int fieldNumber, int length) throws IOException {
+      protected void writeLengthDelimitedField(int fieldNumber, int length) throws IOException {
          writeVarint32(WireType.makeTag(fieldNumber, WireType.WIRETYPE_LENGTH_DELIMITED));
          writeVarint32(length);
       }
 
       // low level ops, writing values without tag
 
-      abstract void writeVarint32(int value) throws IOException;
+      protected abstract void writeVarint32(int value) throws IOException;
+
+      protected abstract void writeVarint64(long value) throws IOException;
+
+      protected abstract void writeFixed32(int value) throws IOException;
 
-      abstract void writeVarint64(long value) throws IOException;
+      protected abstract void writeFixed64(long value) throws IOException;
 
-      abstract void writeFixed32(int value) throws IOException;
+      protected abstract void writeByte(byte value) throws IOException;
 
-      abstract void writeFixed64(long value) throws IOException;
+      protected abstract void writeBytes(byte[] value, int offset, int length) throws IOException;
 
-      abstract void writeByte(byte value) throws IOException;
+      protected abstract void writeBytes(ByteBuffer value) throws IOException;
+
+      protected boolean supportsFixedVarint() {
+         return false;
+      }
+   }
 
-      abstract void writeBytes(byte[] value, int offset, int length) throws IOException;
+   public abstract static class FixedVarintEncoder extends Encoder {
```

Comment:
I don't think this class is required. 
`skipFixedVarint` and `writePositiveFixedVarint` can be implemented by `Encoder` to throw `UnsupportedOperationException`

Reply:
Sure, originally I had an instanceof check before I added the supportsFixedVarint method.